src/ass/to_ass.py

```diff
@@ -1,84 +1,242 @@
 from collections.abc import Sequence, Mapping
 from datetime import timedelta
-from functools import reduce
-import itertools
+from abc import ABC, abstractmethod
 
 import pyass
 
-from src.fonts import FontScaler
-from src.models import Song, SongLine, SongTitle
+from src.models import Song, SongLine
 
 from .consts import *
 from ..models.karaoke import *
-from .kfx import to_default_event, to_shad_event
 
 
-def to_divider_event(song: Song, dividerText: str) -> pyass.Event:
-    return pyass.Event(
-        format=pyass.EventFormat.COMMENT,
-        style=DIVIDER_STYLE.name,
-        end=song.end,
-        text=dividerText,
+class Effect(ABC):
+    @abstractmethod
+    def to_events(
+        self, song: Song, actorToStyle: Mapping[str, Sequence[pyass.Tag]]
+    ) -> Sequence[pyass.Event]:
+        ...
+
+
+class LyricsEffect(Effect):
+    def __init__(
+        self,
+        shouldPrintTitle: bool = True,
+        dividerStyle: pyass.Style = DIVIDER_STYLE,
+        titleStyle: pyass.Style = TITLE_STYLE,
+        end: timedelta = TITLE_EVENT_DURATION,
```

Comment:
Oops, missed this on the previous round of review

`end` -> `titleEventDuration`

Reply:
Fixed.